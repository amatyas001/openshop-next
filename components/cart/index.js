import { CartContent } from './CartContent/CartContent';
import { CartAmount } from './CartAmount/CartAmount';
import { CartButton } from './CartButton/CartButton';
import { CartItem } from './CartItem/CartItem';
import { CartPanel } from './CartPanel/CartPanel';

export { CartContent, CartAmount, CartButton, CartItem, CartPanel };
