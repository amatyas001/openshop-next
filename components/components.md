Elements are composed by [Chakra-UI](https://chakra-ui.com) components and preserve the library features so [Style Props](https://chakra-ui.com/style-props) can be passed to them and accessibility remains optimized.

> _Style Props are a way to alter the style of a component by simply passing props to it. It helps to save time by providing helpful shorthand ways to style components._

For more information visit the [Chakra UI](https://chakra-ui.com) documentation.

Components are divided into two main category: _wrappers_ and _single responsibility_ elements. Wrapper components does not contain business logic, however there are special cases like _handling lifted states to keep sensitive data locally_ and not to populate them to the Redux store which is persisted on the machine.

**State Dependencies** and **Wrapped Components** are documented in each description to give brief information about the context of posible interactions of the actual component. _Examples_ prepared with proper state and may render extra controls to test different cases with ease.
