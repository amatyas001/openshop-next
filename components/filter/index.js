export * from './FilterButton/FilterButton';
export * from './FilterColor/FilterColor';
export * from './FilterPrice/FilterPrice';
export * from './FilterName/FilterName';
export * from './FilterPanel/FilterPanel';
export * from './FilterReset/FilterReset';
