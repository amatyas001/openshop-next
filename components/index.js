export * from './shared';
export * from './cart';
export * from './payment';
