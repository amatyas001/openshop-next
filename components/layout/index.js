export * from './Content/Content';
export * from './Footer/Footer';
export * from './ProductsTopBar/ProductsTopBar';
export * from './Navbar/Navbar';
export * from './ScrollTop/ScrollTop';
