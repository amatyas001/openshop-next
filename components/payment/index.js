export * from './PaymentCancelled/PaymentCancelled';
export * from './PaymentConfirm/PaymentConfirm';
export * from './PaymentConfirmCard/PaymentConfirmCard';
export * from './PaymentConfirmControls/PaymentConfirmControls';
export * from './PaymentConfirmDetails/PaymentConfirmDetails';
export * from './PaymentError/PaymentError';
export * from './PaymentForm/PaymentForm';
export * from './PaymentFormControls/PaymentFormControls';
export * from './PaymentProgress/PaymentProgress';
export * from './PaymentReview/PaymentReview';
export * from './PaymentSuccess/PaymentSuccess';
