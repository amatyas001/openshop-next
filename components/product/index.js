import { ProductCard } from './ProductCard/ProductCard';
import { ProductControls } from './ProductControls/ProductControls';
import { ProductControlsColor } from './ProductControlsColor/ProductControlsColor';
import { ProductControlsAmount } from './ProductControlsAmount/ProductControlsAmount';
import { ProductControlsButton } from './ProductControlsButton/ProductControlsButton';
import { ProductControlsSize } from './ProductControlsSize/ProductControlsSize';

export {
  ProductCard,
  ProductControls,
  ProductControlsColor,
  ProductControlsAmount,
  ProductControlsButton,
  ProductControlsSize,
};
