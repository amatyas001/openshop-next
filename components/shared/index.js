import { Button } from './Button/Button';
import { ButtonIcon } from './ButtonIcon/ButtonIcon';
import { Field } from './Field/Field';
import { Image } from './Image/Image';
import { Panel } from './Panel/Panel';
import { Spinner } from './Spinner/Spinner';

export { Button, ButtonIcon, Field, Image, Panel, Spinner };
