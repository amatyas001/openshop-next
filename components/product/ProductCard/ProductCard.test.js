import { create, act } from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { mockProduct } from '@app/mocks';
import { INITIAL_STATE } from '@app/config';
import { ProductCard } from '@app/components';

let tree;
const product = mockProduct(1)[0];
const mockStore = configureStore([]);

describe('<ProductCard />', () => {
  beforeAll(() => {
    act(() => {
      tree = create(
        <Provider store={mockStore(INITIAL_STATE)}>
          <ProductCard product={product} />
        </Provider>
      );
    });
  });
  it('should render with required props', () => {
    expect.assertions(1);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
