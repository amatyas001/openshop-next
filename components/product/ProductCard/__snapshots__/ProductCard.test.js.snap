// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductCard /> should render with required props 1`] = `
<div
  className="css-7f5i7h"
  id="mock_id_0"
>
  <h2
    className="css-b8gb46"
  >
    mock_name_0
  </h2>
  <div
    className="css-jrczjt"
  >
    <div
      className="css-edqtqw"
      label="Loading content"
      src="/images/products/mock_img_0"
    >
      <h2
        className="css-1gjlikv"
      >
        Loading...
      </h2>
      <div
        className="css-159m1lu"
      >
        <div
          className="css-1gy22ay"
        >
          Loading...
        </div>
      </div>
    </div>
    <img
      className="css-4jmajs"
    />
    <p
      className="css-9ckoz9"
    >
      mock_description_0
    </p>
    <div
      className="css-1j174qx"
    >
      <span />
      <h2
        className="css-d94gct"
      >
        1&nbsp;$
      </h2>
      <p
        className="css-u84yc3"
      >
        mock_color_0
      </p>
      <div
        className="css-1nqhd46"
      >
        <svg
          className="css-1qla8s8"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 576 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
          />
        </svg>
      </div>
      <p
        className="css-u84yc3"
      >
        One Size
      </p>
      <p
        className="css-1nsgbpp"
      >
        1 in stock
      </p>
      <div
        className="css-1dm3qsp"
        data-testid="product-amount-wrapper"
      >
        <div
          className="css-hboir5"
        >
          <div
            aria-hidden={true}
            className="css-1yqqx3k"
          >
            <button
              className="css-fdetwv"
              data-testid="product-amount-increment"
              href="#void"
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              role="button"
              tabIndex="0"
              type="button"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-32 252c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92H92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
                />
              </svg>
            </button>
            <button
              className="css-1fd1i4r"
              data-testid="product-amount-decrement"
              href="#void"
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              role="button"
              tabIndex="0"
              type="button"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zM92 296c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h264c6.6 0 12 5.4 12 12v56c0 6.6-5.4 12-12 12H92z"
                />
              </svg>
            </button>
          </div>
          <input
            aria-invalid={false}
            aria-readonly={true}
            aria-valuemax={1}
            aria-valuemin={0}
            aria-valuenow={0}
            autoComplete="off"
            className="css-s3c5ne"
            data-testid="product-amount-input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            readOnly={true}
            role="spinbutton"
            type="text"
            value={0}
          />
        </div>
      </div>
      <button
        className="css-1e5lh77"
        data-testid="product-buy-button"
        disabled={true}
        href="#void"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex="0"
        type="button"
      >
        buy
      </button>
    </div>
  </div>
</div>
`;
